src/Mvc/src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeatureProvider.cs

```diff
@@ -1,111 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Linq;
-using System.Reflection;
-using Microsoft.AspNetCore.Mvc.ApplicationParts;
-using Microsoft.Extensions.Primitives;
-
-namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
-{
-    /// <summary>
-    /// An <see cref="IApplicationFeatureProvider{TFeature}"/> for <see cref="ViewsFeature"/>.
-    /// </summary>
-    [Obsolete("This type is obsolete and will be removed in a future version. See " + nameof(IRazorCompiledItemProvider) + " for alternatives.")]
-    public class ViewsFeatureProvider : IApplicationFeatureProvider<ViewsFeature>
```

Comment:
For contrast, this is random old extensibility that is a good thing to remove since it will rarely appear in user-code.

Reply:
One concern I do have, is if I remove this but don't remove the `RazorViewAttribute` and a user doesn't heed the obsolete warnings they'll get a hard to resolve, significant behavior change.

Reply:
> and a user doesn't heed the obsolete warnings they'll get a hard to resolve

We can add it to the migration notes as something to be aware of. If we get lots of user feedback (I'd think it's fairly unlikely given this attribute comes from Precompilation), we can always change the runtime to throw.

Reply:
I'm still not convinced since i'd rather break hard or not at all then break soft when it comes to these things (can have migration guidance with either hard or soft breaks). I'll leave it as removed though since you both feel differently 😄 

@danroth27 is there a place where we're collecting migration guidance?